src/metabase/task/search_index.clj

```diff
@@ -11,22 +11,32 @@
 
 (set! *warn-on-reflection* true)
 
-;; We need each instance to initialize on start-up currently, this will need to be refined.
-(jobs/defjob ^{DisallowConcurrentExecution false
+(def ^:private recreated? (atom false))
+
+(def job-key
+  "Key used to define and trigger the search-index task."
+  (jobs/key "metabase.task.search-index.job"))
+
+(jobs/defjob ^{DisallowConcurrentExecution true
                :doc                        "Populate Search Index"}
   SearchIndexing [_ctx]
-  (when (public-settings/experimental-fulltext-search-enabled)
-    (log/info "Recreating search index from latest schema")
-    (search/init-index! {:force-reset? true})
-    (log/info "Populating search index")
-    (search/reindex!)
```

Comment:
This reindexing was actually redundant with initialization, I was just kicking the tires.